skimage/_build.py

```diff
@@ -24,6 +24,11 @@ def cython(pyx_files, working_path=''):
     if len(sys.argv) >= 2 and sys.argv[1] == 'clean':
         return
 
+    cython_options = {}
+    if os.environ.get('CACHE_CYTHON', 'False').lower() == 'true':
```

Comment:
I tried doing this on my local machine, and it seems like caching is does not work for older versions of Cython, and appears to be broken in Python 3, due to this line: https://github.com/cython/cython/blob/master/Cython/Build/Dependencies.py#L542.